### DISCUSSION TITLE
Numerical instability issue

### DISCUSSION THREAD

**Discussion by MNJOLE** [firemodels/fds#13008] [Q&A]
Hello, I am currently facing an unknown numerical instability issue.

I am trying to analyze the combustion characteristics of PMMA using a cone heater. The setup is such that PMMA generates MMA during combustion, and MMA gas is set to react.

I have reviewed the setup as described in the user guide:

1. The cell size of the mesh is always applied uniformly with a width-to-height ratio of 1:1. (Numerical instability occurred both when cell size factor and stretch factor were applied to the PMMA surface and when they were not applied.)
2. The cell size is uniform in all areas where the mesh is applied.
3. All areas are designated as open, and a duct with a flow rate of 0.024 m3/s is set at the top. The flow rate is not very high, and since the space is open and not compartmentalized, it does not seem to cause pressure issues.
4. There is no sudden change in the heat release rate due to the combustion simulation of PMMA.
5. Both INERT and ADIABATIC were set outside the cone heater, but the same error occurred.
6. No changes were made to the space configuration (all open).
7. Although MMA is a polymer, the specific heat of the gas fuel was specified in the SPEC line as described in the user guide. (C5H8O2)
8. I suspected the error might be due to the time step, so I adjusted it very finely to dt=0.005, but the error was not resolved.

I'm currently using FDS 6.9.1 version.
I would really appreciate it if you could let me know what else I need to review.

**Comment by drjfloyd** [comment]
Try the current test release (https://github.com/firemodels/test_bundles). There has been some recent work on the numerical stability for pyrolyzing surfaces.

**Reply by MNJOLE** [reply]
SMV-6.9.4-0-g526c020 2024-May-29 11:29
Is it correct to download this version and try it?
I'm sorry that I'm not good at English. I don't know if I understood it correctly..

**Reply by TristanHehnen** [reply]
SMV should be Smokeview. If you scroll down a bit, there is an entry for FDS: "[FDS-6.9.1-302-gabbe496 2024-May-28 14:53](https://github.com/firemodels/test_bundles/releases/tag/FDS_TEST)". I assume this is what is meant above.

There was also some discussion a while ago that the `MINIMUM_LAYER_THICKNESS` may need to be adjusted. If I remember correctly, the reasoning was that during the pyrolysis the final solid cell may lead to a sudden spike in HRR when it vanishes.

**Reply by MNJOLE** [reply]
Thank you for explaining clearly. 
I will read the user guide for the input section more thoroughly. 
I have an additional question: When implementing the combustion of a specimen in a cone heater, I intend to apply burnaway.
Would this attempt align with my intentions? Also, could the numerical instability be caused by burnaway? 
I would like your opinion on this.

**Reply by TristanHehnen** [reply]
I'm not familiar with how `BURN_AWAY` works deep down. Depending on how FDS initialises the freed gas cell and its boundaries, once the `OBST` disappears, there could be issues. 

For now, I've only conducted cone calorimeter simulations with the "classical" 1d heat transfer `SURF` and no `BURN_AWAY`. As sample PMMA with a thickness of 6 mm was used. The highest fluid cell resolution was 2 mm and the heater was created from `GEOM`. I've also observed numerical instabilities towards the end of the simulations (FDS 6.8.0).
I did use fairly small solid cells (i.e. `SURF`), with a `CELL_SIZE_FACTOR = 0.1`, `STRETCH_FACTOR = 1.0` and `WALL_INCREMENT = 1`. The `CELL_SIZE_FACTOR = 0.1` was excessive, because it consumes a lot of computing time.

**Reply by drjfloyd** [reply]
Tristan has identified the file. Whatever the latest FDS-SMV install is in that repo. 

When you have a pyrolyzing surface there are two things that lead to numerical challenge. First, if there is no residue, then the surface can go to zero thickness. That leads to numerical challenges with the in-depth heat transfer. A method to limit that potential impact is to pick some MINIMUM_LAYER_THICKNESS where we just remove the layer and add its mass to the gas phase. The challenge there is depending on grid size and timestep, the implied surface velocity can result in instabilities. There has been more work on this since 6.9.1 so the latest version might avoid the instability. If not you can reduce the value of MINIMUM_LAYER_THICKESS on the SURF input.
